bot/cogs/assignable_roles_cog.py

```diff
@@ -19,25 +21,139 @@ def __init__(self, bot) -> None:
+                emojis = ''
+            names = names + '\n' + emojis + str(matching_roles[k])
+        
+        embed = discord.Embed(title= title, color= Colors.ClemsonOrange)
+        embed.add_field(name= 'Matching Roles:', value= names + choose)
+
+        mes: Message = await ctx.send(embed= embed)
+        
+        # Remove unnecessary emojis
+        if role_count < len(reactions):
+            #print('\n' + str(len(reactions)) + '\n')
+            del reactions[role_count:len(reactions)]
+
+        """
+            USING EMOJIS WITH d.py
+                NOTE: This only works for UNICODE emojis. I have no idea how to get DISCORD specific emojis to work
+
+                Guide: https://medium.com/@codingpilot25/how-to-print-emojis-using-python-2e4f93443f7e
+                A list of emojis: https://www.unicode.org/emoji/charts/emoji-list.html
+                
+                Additional info: https://docs.python.org/3/reference/lexical_analysis.html#string-and-bytes-literals
+
+                Method 1:
+                say you wants :ping_pong:, you would use the unicode charcter U+1F3D3 and change it to U0001F3D3.
+                    thus, '\U0001F3D3' would be your string for ping pong
+                as another example, if you wanted to use :skull_crossbones:, you would use unicode character U+2620 and change it to U00002620
+                    thus, '\U00002620' would be your string for skull and crossbones
+
+                Method 2:
+                Use the CDLR short names.
+                Not recommended. Only worked if name contains ONLY letters and spaces.
+        """
+                #e.g. '\N{pensive face}' is fine but '\N{keycap: 1} would NOT work'
+
+        # Add reactions for user to choose from
+        for str_emoji in reactions:
+            await mes.add_reaction(str_emoji)
+
+        # Validate the answer using a reaction event loop.
+        def predicate(reaction: Reaction, user: Member) -> bool:
+            # Test if the the answer is valid and can be evaluated.
+            return (
+                reaction.message.id == mes.id         # The reaction is attached to the question we asked.
+                and user == ctx.author                # It's the user who triggered the quiz.
+                and str(reaction.emoji) in reactions  # The reaction is one of the options.
+            )
+
+        try:
+            reaction, user = await ctx.bot.wait_for("reaction_add", timeout=90.0, check=predicate)
+        except asyncio.TimeoutError: # was asyncio.
+            await ctx.send(f'You took too long.')
+            await mes.clear_reactions()
             return
         
-        await self.set_role(ctx, role)
+        answer = -1
+        
+        for m in range(len(reactions)):
+            if str(reaction.emoji) == reactions[m]:
+                answer = m
+        
+        if answer > -1:
+            await ctx.send(f'User reaction was associated with role **{reactions[answer]}**.')
```

Comment:
we should swap 138 and 139, so that if what whatever reason adding a role throws something random we dont send a false "role added" message

Reply:
The message was meant to be more of an:
>FYI this is the role associated with the emoji you reacted with

IMO it doesn't need to be here at all, since the `set_role` command will provide the same information once it runs. I'm commenting it out for now.

Reply:
fair enough, we can prob just delete it